### ISSUE TITLE
add ocean variable list

### ISSUE THREAD

**Issue by larsbuntemeyer** [WCRP-CORDEX/data-request-table#33]
from https://zenodo.org/records/8207553

**Comment by larsbuntemeyer** [comment]
@gnikulin is this for all domains, or actually only for MED-CORDEX?

**Comment by gnikulin** [comment]
it's only for Med-CORDEX so we need to think first how to organise domain-specific DRs

**Comment by sol1105** [comment]
The ocean data request got an update: https://zenodo.org/records/13165021

Is it expected that this or a similar variable list will be added to the CORDEX-CMIP6 CMOR tables so that the ocean variables from coupled CORDEX-CMIP6 runs (there are several coupled models registered with the CV, i.e. with `"source_type": "AORCM"`)  can be published via the same ESGF project as the atmosphere data?

DWD-BSH have conducted first CORDEX-CMIP6 conformal model simulations with the model `ROAM-NBS` (Regional coupled ocean-atmosphere model based on icon-2024.07 and NEMO4.2.0 for the North and Baltic Sea) and while the atmosphere data can be published via the CORDEX-CMIP6 ESGF project (and soon will be) I wonder what to do with the ocean (and sea ice) data.

**Comment by jesusff** [comment]
Hi Martin,

yes, this is pending, and should be handled on time for the ESGF upload. Note that publishing CORDEX-CMIP6 data on ESGF is not yet possible. We are also working on that at the moment (https://github.com/WCRP-CORDEX/cordex-cmip6-cv/issues/165). I assume you are adopting the Med-CORDEX variable list for the ocean. As far as I know, this has not been discussed yet in the EURO-CORDEX community, but could be a good topic for the upcoming GA (@ClaasTeichmann).

Sea ice variables might be a problem, as these are not considered in Med-CORDEX. If you have a proposed data request, do not hesitate to open a specific issue or PR for discussion. All this might be handled by the TF-Ocean (@sam-somot) at some point also. 

Thanks!

**Comment by larsbuntemeyer** [comment]
I got an excel sheet once from @gnikulin which i converted and still had on my disk. I added it in #52 
However, this still needs a little cleaning, but it's a join of the excel sheet with CMIP6 cmor tables to get missing meta data like cell_methods. It would be good if we could handle additional variables through PRs on our [main table](https://github.com/WCRP-CORDEX/data-request-table/blob/main/cmor-table/datasets.csv) in the future. These google spreadsheets usually miss a lot of crucial meta data.

**Comment by larsbuntemeyer** [comment]
Should the MED-CORDEX data request be like default data requets + ocean variables? Or should ocean variable be added to the default?

**Comment by jesusff** [comment]
A bit more complicated due to the priority being dependent on whether a given model has a given component (ocean, river , etc). There are also data requests for rivers (https://doi.org/10.5281/zenodo.7112672) and aerosols (https://doi.org/10.5281/zenodo.7112859) that should be included in the Med-CORDEX data request. We should be able to get the corresponding excel files from @sam-somot

At some point, I think we decided to have a dummy column "realm" to pile up these different data requests into a single file for Med-CORDEX. The priorities in them are only meaningful if the model has the corresponding component (i.e. contributes to that realm). We could use the realms from CMIP6 (https://github.com/WCRP-CMIP/CMIP6_CVs/blob/main/CMIP6_realm.json), however, there is no river realm. At some point they considered model components (https://github.com/WCRP-CMIP/CMIP6_CVs/issues/604) but dropped the idea. We could use a word different from "realm" to avoid confusion if our realms do not match those in CMIP6.

Anyway, this doesn't mean that the realms are back to CORDEX CV, as the priority and realm columns drop completely in the CMOR tables and the DRS. We discussed this somewhere, but I cannot find it now...

> Should the MED-CORDEX data request be like pd.concat([default data requets, ocean variables])?

We should also make sure that the atmos part is the same as the CORDEX default. Hard to see from the PDF. The Med-CORDEX one has date 2022-12-01 (maybe it is just the date when it was approved by Med-CORDEX, @sam-somot?), while the general CORDEX one is from 2022-09-22.

**Comment by larsbuntemeyer** [comment]
The current ocean variable list has a couple of unregistered standard names:
```
['gibraltar_heat_flux_in',
 'gibraltar_heat_flux_net',
 'gibraltar_salt_flux_in',
 'gibraltar_salt_flux_net',
 'gibraltar_water_flux_in',
 'gibraltar_water_flux_net',
 'heat_flux_correction',
 'sea_water_z_velocity',
 'surface_x_velocity',
 'surface_y_velocity']
```

I guess the following:

* `wo`: `sea_water_z_velocity` should be `upward_sea_water_velocity`
* `uos`: `surface_x_velocity` should be `eastward_sea_water_velocity`
* `vos`: `surface_y_velocity` should be `northward_sea_water_velocity`
* `heat_flux_correction`: This seems actually be an alias for `heat_flux_into_sea_water_due_to_flux_adjustment`, so this should be fine.

**Comment by larsbuntemeyer** [comment]
There is something comparable like [mfo](https://github.com/PCMDI/cmip6-cmor-tables/blob/6a30acaea9b63fdecbe08894d7e50ef36879a993/Tables/CMIP6_Omon.json#L2897-L2914) with a coordinate [oline](https://github.com/PCMDI/cmip6-cmor-tables/blob/6a30acaea9b63fdecbe08894d7e50ef36879a993/Tables/CMIP6_coordinate.json#L984-L1023) that defines an integral along a line. Maybe that's the way to go for these strait fluxes? There is actually a definition `"gibraltar_strait"`.

**Comment by larsbuntemeyer** [comment]
I'm not unsure about `tauuo` / `tauvo`: In CMIP6 cmor tables they use this out_name for `downward_x_stress_at_sea_water_surface` / `Sea Water Surface Downward X Stress` while the `surface_downward_x_stress` / `X-Component of Atmospheric Stress on Sea Ice` has out_name `sistrxdtop`. I gues that's not what MED-CORDEX had in mind. So i would also suggest to change standard names and long names of `tauuo` / `tauvo` to be consisten with CMIP6.

This is the overall join of ocean variables with cmip6 tables where standard names differ or are missing:

|    | out_name   | standard_name_cmip6                    | standard_name_med           | cmip6_table   | long_name_med                                    | long_name_cmip6                     |
|---:|:-----------|:---------------------------------------|:----------------------------|:--------------|:-------------------------------------------------|:------------------------------------|
| 40 | wo         | upward_sea_water_velocity              | sea_water_z_velocity        | Odec          | Sea Water Upward Velocity                        | Sea Water Vertical Velocity         |
| 41 | wo         | upward_sea_water_velocity              | sea_water_z_velocity        | Omon          | Sea Water Upward Velocity                        | Sea Water Vertical Velocity         |
| 42 | rhop       | nan                                    | sea_water_potential_density | nan           | Sea Water Potential Density                      | nan                                 |
| 51 | tauuo      | downward_x_stress_at_sea_water_surface | surface_downward_x_stress   | Odec          | Surface Downward X Stress                        | Sea Water Surface Downward X Stress |
| 52 | tauuo      | downward_x_stress_at_sea_water_surface | surface_downward_x_stress   | Omon          | Surface Downward X Stress                        | Sea Water Surface Downward X Stress |
| 53 | tauuo      | downward_x_stress_at_sea_water_surface | surface_downward_x_stress   | Odec          | Surface Downward X Stress                        | Sea Water Surface Downward X Stress |
| 54 | tauuo      | downward_x_stress_at_sea_water_surface | surface_downward_x_stress   | Omon          | Surface Downward X Stress                        | Sea Water Surface Downward X Stress |
| 55 | tauvo      | downward_y_stress_at_sea_water_surface | surface_downward_y_stress   | Odec          | Surface Downward Y Stress                        | Sea Water Surface Downward Y Stress |
| 56 | tauvo      | downward_y_stress_at_sea_water_surface | surface_downward_y_stress   | Omon          | Surface Downward Y Stress                        | Sea Water Surface Downward Y Stress |
| 57 | tauvo      | downward_y_stress_at_sea_water_surface | surface_downward_y_stress   | Odec          | Surface Downward Y Stress                        | Sea Water Surface Downward Y Stress |
| 58 | tauvo      | downward_y_stress_at_sea_water_surface | surface_downward_y_stress   | Omon          | Surface Downward Y Stress                        | Sea Water Surface Downward Y Stress |
| 59 | uos        | nan                                    | surface_x_velocity          | nan           | Surface X Velocity                               | nan                                 |
| 60 | vos        | nan                                    | surface_y_velocity          | nan           | Surface Y Velocity                               | nan                                 |
| 69 | gibfx      | nan                                    | gibraltar_water_flux_net    | nan           | Water Transport through Gibraltar Strait         | nan                                 |
| 70 | gibfx      | nan                                    | gibraltar_water_flux_net    | nan           | Water Transport through Gibraltar Strait         | nan                                 |
| 71 | gibfxin    | nan                                    | gibraltar_water_flux_in     | nan           | Eastward Water Transport though Gibraltar Strait | nan                                 |
| 72 | gibfxin    | nan                                    | gibraltar_water_flux_in     | nan           | Eastward Water Transport though Gibraltar Strait | nan                                 |
| 73 | gibsf      | nan                                    | gibraltar_salt_flux_net     | nan           | Net Salt Transport through Gibraltar Strait      | nan                                 |
| 74 | gibsf      | nan                                    | gibraltar_salt_flux_net     | nan           | Net Salt Transport through Gibraltar Strait      | nan                                 |
| 75 | gibsfin    | nan                                    | gibraltar_salt_flux_in      | nan           | Eastward Salt Transport through Gibraltar Strait | nan                                 |
| 76 | gibsfin    | nan                                    | gibraltar_salt_flux_in      | nan           | Eastward Salt Transport through Gibraltar Strait | nan                                 |
| 77 | gibhf      | nan                                    | gibraltar_heat_flux_net     | nan           | Net Heat Transport through Gibraltar Strait      | nan                                 |
| 78 | gibhf      | nan                                    | gibraltar_heat_flux_net     | nan           | Net Heat Transport through Gibraltar Strait      | nan                                 |
| 79 | gibhfin    | nan                                    | gibraltar_heat_flux_in      | nan           | Eastward Heat Transport through Gibraltar Strait | nan                                 |
| 80 | gibhfin    | nan                                    | gibraltar_heat_flux_in      | nan           | Eastward Heat Transport through Gibraltar Strait | nan                                 |
| 87 | wo         | upward_sea_water_velocity              | sea_water_z_velocity        | Odec          | Sea Water Upward Velocity                        | Sea Water Vertical Velocity         |
| 88 | wo         | upward_sea_water_velocity              | sea_water_z_velocity        | Omon          | Sea Water Upward Velocity                        | Sea Water Vertical Velocity         |
| 89 | rhop       | nan                                    | sea_water_potential_density | nan           | Sea Water Potential Density                      | nan                                 |


sistrxdtop

**Comment by jesusff** [comment]
> There is something comparable like [mfo](https://github.com/PCMDI/cmip6-cmor-tables/blob/6a30acaea9b63fdecbe08894d7e50ef36879a993/Tables/CMIP6_Omon.json#L2897-L2914) with a coordinate [oline](https://github.com/PCMDI/cmip6-cmor-tables/blob/6a30acaea9b63fdecbe08894d7e50ef36879a993/Tables/CMIP6_coordinate.json#L984-L1023) that defines an integral along a line. Maybe that's the way to go for these strait fluxes? There is actually a definition "gibraltar_strait".

yes, "gibraltar_strait" is a [standardized region name](https://cfconventions.org/Data/cf-standard-names/docs/standardized-region-names.html) in CF. Others that might be of interest are not defined (e.g. there's no Bosphorus strait)

**Comment by larsbuntemeyer** [comment]
> There is something comparable like [mfo](https://github.com/PCMDI/cmip6-cmor-tables/blob/6a30acaea9b63fdecbe08894d7e50ef36879a993/Tables/CMIP6_Omon.json#L2897-L2914) with a coordinate [oline](https://github.com/PCMDI/cmip6-cmor-tables/blob/6a30acaea9b63fdecbe08894d7e50ef36879a993/Tables/CMIP6_coordinate.json#L984-L1023) that defines an integral along a line. Maybe that's the way to go for these strait fluxes? There is actually a definition `"gibraltar_strait"`.

Availabel measures across line are:
```
['ocean_volume_transport_across_line',
 'sea_ice_area_transport_across_line',
 'sea_ice_transport_across_line',
 'sea_water_transport_across_line',
 'snow_transport_across_line_due_to_sea_ice_dynamics']
```
There is nothing salt nor heat related.

**Comment by sol1105** [comment]
Currently new CF standard_names proposals for CMIP7 are discussed in the CF vocabulary github repository issue threads. Among them also `ocean_*_transport_across_line` for salt mass, heat, volume etc. in the following issue thread: https://github.com/cf-convention/vocabularies/issues/218

**Comment by larsbuntemeyer** [comment]
> Currently new CF standard_names proposals for CMIP7 are discussed in the CF vocabulary github repository issue threads. Among them also `ocean_*_transport_across_line` for salt mass, heat, volume etc. in the following issue thread: [cf-convention/vocabularies#218](https://github.com/cf-convention/vocabularies/issues/218)

Great, that looks like what we also would need here!

**Comment by larsbuntemeyer** [comment]
@vmaurerDWD This is the discussion about ocean variables

**Comment by sam-somot** [comment]
@larsbuntemeyer @jesusff : I'm back to the conversation on the ocean variable lists for Med-CORDEX and for a wider use in order to prepare the ESGF publication. Do you need any information or files from me ?

**Comment by larsbuntemeyer** [comment]
I think, this is fine for now and i will merge it into the cmor tables soon.

**Comment by sam-somot** [comment]
I had a recent request for a variable we forgot: SBT : sea bottom temperature. Do you know if there is a CMIP6 standard name for this ocean variable ?

**Comment by larsbuntemeyer** [comment]
There is `sea_water_temperature_at_sea_floor` and i guess the variable name should be `tbo` consistent with `pbo`

**Comment by jesusff** [comment]
But this variable does not exist in CMIP6 https://clipc-services.ceda.ac.uk/dreq/index/CMORvar.html

**Comment by larsbuntemeyer** [comment]
That's true, we should be careful with the `out_name`. There seems to be no clear convention. @sam-somot Can you confirm, you do *not* want, e.g.,  [sea_water_potential_temperature_at_sea_floor](https://github.com/PCMDI/cmip6-cmor-tables/blob/e3644d3b814d632c4343b9ae5a59faf1ed20191c/Tables/CMIP6_Omon.json#L4607-L4624)?

**Comment by sam-somot** [comment]
you are right @larsbuntemeyer , this is what I want the **sea_water_potential_temperature_at_sea_floor** as the prognostic variable in the ocean model is the potential tempeature. Sorry for not detecting this point in the first message

**Comment by larsbuntemeyer** [comment]
Ok, agrred, I'll add that to #52. Please also let me know which frequencies!

**Comment by sam-somot** [comment]
I would go for monthly-mean in TIER1 and daily-mean in TIER2 as it is clearly not CORE
sam

**Comment by larsbuntemeyer** [comment]
I'll close this for now due to https://github.com/WCRP-CORDEX/data-request-table/pull/52.
If some updates are required, this should be discussed in a new issue.
